sct-commons/src/main/java/org/lfenergy/compas/sct/commons/scl/icd/IcdHeaderAdapter.java

```diff
@@ -0,0 +1,81 @@
+        TCompasIEDType iedType,
+        BigInteger iedSubstationinstance,
+        BigInteger iedSystemVersioninstance,
+        String iedName,
+        String vendorName,
+        String iedModel,
+        TCompasIEDRedundancy iedRedundancy,
+        String bayLabel,
+        String hwRev,
+        String swRev,
+        String headerId,
+        String headerVersion,
+        String headerRevision
+) {
+    public IcdHeaderAdapter(TCompasICDHeader compasICDHeader) {
+        this(compasICDHeader.getContent(), compasICDHeader.getICDSystemVersionUUID(), compasICDHeader.getIEDType(),
+                compasICDHeader.getIEDSubstationinstance(), compasICDHeader.getIEDSystemVersioninstance(), compasICDHeader.getIEDName(),
+                compasICDHeader.getVendorName(), compasICDHeader.getIEDmodel(),
+                compasICDHeader.getIEDredundancy(), compasICDHeader.getBayLabel(), compasICDHeader.getHwRev(), compasICDHeader.getSwRev(),
+                compasICDHeader.getHeaderId(), compasICDHeader.getHeaderVersion(), compasICDHeader.getHeaderRevision());
+    }
+
+    public TCompasICDHeader toTCompasICDHeader() {
+        TCompasICDHeader tCompasICDHeader = new TCompasICDHeader();
+        tCompasICDHeader.setContent(content);
+        tCompasICDHeader.setICDSystemVersionUUID(icdSystemVersionUUID);
+        tCompasICDHeader.setIEDType(iedType);
+        tCompasICDHeader.setIEDSubstationinstance(iedSubstationinstance);
+        tCompasICDHeader.setIEDSystemVersioninstance(iedSystemVersioninstance);
+        tCompasICDHeader.setIEDName(iedName);
+        tCompasICDHeader.setVendorName(vendorName);
+        tCompasICDHeader.setIEDmodel(iedModel);
+        tCompasICDHeader.setIEDredundancy(iedRedundancy);
+        tCompasICDHeader.setBayLabel(bayLabel);
+        tCompasICDHeader.setHwRev(hwRev);
+        tCompasICDHeader.setSwRev(swRev);
+        tCompasICDHeader.setHeaderId(headerId);
+        tCompasICDHeader.setHeaderVersion(headerVersion);
+        tCompasICDHeader.setHeaderRevision(headerRevision);
+        return tCompasICDHeader;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        IcdHeaderAdapter that = (IcdHeaderAdapter) o;
+        return Objects.equals(icdSystemVersionUUID, that.icdSystemVersionUUID) && iedType == that.iedType && Objects.equals(vendorName, that.vendorName) && Objects.equals(iedModel, that.iedModel) && iedRedundancy == that.iedRedundancy && Objects.equals(hwRev, that.hwRev) && Objects.equals(swRev, that.swRev) && Objects.equals(headerId, that.headerId) && Objects.equals(headerVersion, that.headerVersion) && Objects.equals(headerRevision, that.headerRevision);
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(icdSystemVersionUUID, iedType, vendorName, iedModel, iedRedundancy, hwRev, swRev, headerId, headerVersion, headerRevision);
+    }
```

Comment:
It would be more explicit which field is excluded from equals and hashCode if you use Lombok @EqualsAndHashCode and @EqualsAndHashCode.Exclude.

Reply:
Yes of course, didn't remember we could use lombok in compas :sweat_smile:

Reply:
not working on records